src/config.rs

```diff
@@ -166,25 +169,61 @@ impl From<LdapConfig> for ldap_poller::Config {
 #[derive(Debug, Clone, Deserialize)]
 pub struct LdapAttributesMapping {
 	/// Attribute for the user's first name
-	pub first_name: String,
+	pub first_name: AttributeMapping,
 	/// Attribute for the user's last name
-	pub last_name: String,
+	pub last_name: AttributeMapping,
 	/// Attribute for the user's preferred username
-	pub preferred_username: String,
+	pub preferred_username: AttributeMapping,
 	/// Attribute for the user's email address
-	pub email: String,
+	pub email: AttributeMapping,
 	/// Attribute for the user's phone number
-	pub phone: String,
+	pub phone: AttributeMapping,
 	/// Attribute for the user's unique ID
-	pub user_id: String,
+	pub user_id: AttributeMapping,
 	/// This attribute shows the account status (LDAP = Enabled, accountStatus)
-	pub status: String,
+	pub status: AttributeMapping,
 	/// Marks an account as enabled (LDAP = TRUE, active)
 	pub enable_value: String,
 	/// Marks an account as enabled (LDAP = FALSE, inactive)
 	pub disable_value: String,
```

Comment:
`StringOrBytes` for these maybe? Since `status` can be binary

Reply:
The value of the status attribute can be a binary when received from ldap, but these values can never be binaries since they come from our config file. So I think this represents the actual type of these values better.

Reply:
Ok, I've explicitly implemented `PartialEq` so it will compare the bytes as well, in case the status values are in fact binary.

Reply:
what if `enable_value` is some bytestring non-representable as utf? Is it possible to specify invalid utf in yaml/json and make `serde_json` swallow it?

Reply:
I don't really think so. We *could* add explicit support for this, by parsing numbers to `Vec<u8>` if that specific attribute is supposed to be binary, so that users can still represent such values.

That said, in practice this isn't the case for AD or openldap at the moment. I'd say we should hold off on supporting this until we at least have an actual use case.